zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -799,4 +802,66 @@ public void testAsyncWriteOperations() {
       zkClient.delete("/tmp/asyncOversize");
     }
   }
+
+  /*
+   * Tests getChildren() when there are an excessive number of children and connection loss happens,
+   * the operation should terminate and exit retry loop.
+   */
+  @Test
+  public void testGetChildrenOnLargeNumChildren() throws Exception {
+    // Default packetLen is 4M. It is static final and initialized
```

Comment:
I don't think it is necessary, because anyway it checks jute.maxbuffer. What you mean 10K children nodes are for 4 MB, it is still defined by jute.maxbuffer. We adjust this value to smaller one for testing so only 5 children are needed. Either 100K or 5, it is behaving the same, why would we bother to create 10K children, right? :)